libraries/AP_AIS/AP_AIS.cpp

```diff
@@ -566,63 +570,21 @@ bool AP_AIS::decode_position_report(const char *payload, uint8_t type)
         return true;
     }
 
-    // mask of flags that we update in this message
-    const uint16_t mask = ~(AIS_FLAGS_POSITION_ACCURACY | AIS_FLAGS_VALID_COG | AIS_FLAGS_VALID_VELOCITY | AIS_FLAGS_VALID_TURN_RATE | AIS_FLAGS_TURN_RATE_SIGN_ONLY);
-    uint16_t flags = _list[index].info.flags & mask; // clear all flags that will be updated
-
     // Position accuracy
-    if (pos_acc) {
-        flags |= AIS_FLAGS_POSITION_ACCURACY;
-    }
+    _list[index].set_pos_acc(pos_acc);
 
     // Course over ground
-    if (cog < 36000) {
-        flags |= AIS_FLAGS_VALID_COG;
-    } else {
-        // zero for invalid
-        cog = 0;
```

Comment:
we've lost this ==0 invalid check. The new handler never checks for 0. Also, if 0 is invalid, why is 36000 invalid? I would think it would include that? Is that how the spec is?

Reply:
The AIS spec says `Course over ground will be 3600 (0xE10) if that data is not available.` That is in 0.1 degrees we have already converted to cd here hence the `36000`. The MAVLink spec dons't give a specific value for invalid. Instead it has the flag.